src/basic/confidential-virt.c

```diff
@@ -226,7 +226,18 @@ static ConfidentialVirtualization detect_confidential_virtualization_impl(void)
 
         return CONFIDENTIAL_VIRTUALIZATION_NONE;
 }
+#elif defined(__aarch64__)
+static ConfidentialVirtualization detect_confidential_virtualization_impl(void) {
+        _cleanup_free_ char *s = NULL;
+        size_t readsize;
+        int r;
 
+        r = read_full_virtual_file("/sys/devices/platform/arm-cca-dev/modalias", &s, &readsize);
```

Comment:
hmm, why read modalias? why not check `access("/sys/devices/platform/arm-cca-dev/", F_OK);`? that would check that the dir exists, and that should be enough I presume, no?

also, please debug log about the errors you receieve here, even if you ignore them.

Reply:
I considered just checking for the dir, but went with modalias, either should work fine.

Sure, on the debug logging.

Reply:
the access() call is much lighter, an less likely to run into permission problems.

and who knows, if people build kernels without kmod, aybe the moadlias goes missing... hence, i'd just check for the dir, seems a lot more robust and simpler and faster.

Reply:
This change, and the man pages should now be updated.